### PR TITLE
Problem: currently used go-ethereum contains a known vulnerability

### PR DESCRIPTION
**Pull request by yihuang**
Closes: #142 #102

Solution:
- update to ethermint's recent main branch
- add feemarket module, have to do this together.

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [x] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [x] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [x] Have you rebased your work on top of the latest master? 
- [x] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -40,6 +40,3 @@ replace github.com/cosmos/ibc-go => github.com/crypto-org-chain/ibc-go v1.0.1-ho
 replace github.com/peggyjv/gravity-bridge/module => github.com/crypto-org-chain/gravity-bridge/module v0.1.22-0.20211004080324-36ed315b3281
 
 replace github.com/cosmos/iavl => github.com/cosmos/iavl v0.17.1
```
Comment (tomtau) [inline go.mod:42]: is this IAVL line necessary? it should be by default in 0.44.1?

**Review by tomtau — CHANGES_REQUESTED** [review]
probably the more precise problem is: "Problem: currently used go-ethereum contains a known vulnerability" :) 

I guess also a changelog entry? 

If the current testnet is to be upgraded via a state export -- I guess it'd also need genesis migrations to add the new module?

File: go.mod (hunk 1 above)
Comment (yihuang) [reply go.mod:42]: https://github.com/tendermint/tendermint/blob/v0.34.13/go.mod#L16

it seems still needed, tendermint depends on the old iavl

**Comment by thomas-nguy** [comment]
Let's wait until we have a more stable version of ethermint before upgrading and adding a Changelog entry? (unless we are planning release soon)

It seems the bump to 1.10.9 is still work in progress
https://github.com/tharsis/ethermint/pull/624

File: go.mod (hunk 1 above)
Comment (JayT106) [reply go.mod:42]: submitted https://github.com/tendermint/tendermint/issues/7089, I think it can be backported to v0.34.x.

File: go.mod [hunk 2]
```diff
@@ -41,5 +41,4 @@ replace github.com/peggyjv/gravity-bridge/module => github.com/crypto-org-chain/
...
 
-// FIXME: https://github.com/crypto-org-chain/ethermint/tree/cronos2
-replace github.com/tharsis/ethermint => github.com/crypto-org-chain/ethermint v0.4.2-0.20211004101819-7b5449f36cc8
+replace github.com/tharsis/ethermint => github.com/crypto-org-chain/ethermint v0.4.2-0.20211012020956-6c537e87c727
```
Comment (yihuang) [inline go.mod:44]:
```suggestion
// FIXME: https://github.com/crypto-org-chain/ethermint/tree/nil-tracer
replace github.com/tharsis/ethermint => github.com/crypto-org-chain/ethermint v0.4.2-0.20211012020956-6c537e87c727
```

Comment (yihuang) [reply go.mod:44]: https://github.com/tharsis/ethermint/pull/660

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,8 @@
 
 ### Improvements
 - [cronos#162](https://github.com/crypto-org-chain/cronos/pull/162) bump ibc-go to v1.2.1 with hooks support
+- [cronos#143](https://github.com/crypto-org-chain/cronos/pull/143) update go-ethereum to 1.10.9 and add fee market
```
Comment (thomas-nguy) [inline CHANGELOG.md:12]: maybe specify upgrade ethermint

File: scripts/cronos-devnet.yaml [hunk 4]
```diff
@@ -26,6 +26,8 @@ cronos_777-1:
       evm:
         params:
           evm_denom: basetcro
+          chain_config:
+            london_block: "9223372036854775808"
```
Comment (thomas-nguy) [inline scripts/cronos-devnet.yaml:30]: does it work with nil?

File: CHANGELOG.md [hunk 5]
```diff
@@ -9,6 +9,8 @@
```
Comment (tomtau) [inline CHANGELOG.md:9]: not sure if changelog should also comment on breaking changes?

Comment (yihuang) [reply CHANGELOG.md:11]: copied some ethermint changelog.

File: CHANGELOG.md (hunk 3 above)
Comment (yihuang) [reply CHANGELOG.md]: done

**Comment by yihuang** [comment]
ethermint has upgraded, we just need to update to ethermint main branch later.

**Pull request closed without merging**